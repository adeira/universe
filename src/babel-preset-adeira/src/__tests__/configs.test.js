// @flow strict

import path from 'path';

import preset from '../index';

expect.addSnapshotSerializer({
  test: () => true,
  print: value => {
    const cwd = path.join(__dirname, '..', '..'); // root of our Babel Preset
    return JSON.stringify(value, null, 2).replace(new RegExp(cwd, 'g'), '<BABEL_PRESET_ROOT>');
  },
});

const supportedTargets = ['js', 'js-esm', 'flow'];
const environments = [
  undefined, // default values
  { node: 'current' },
  { browsers: ['last 2 versions', 'ie >= 11'] },
];

const matrix = [];
supportedTargets.forEach(function(target) {
  environments.forEach(function(environment) {
    matrix.push([target, environment]);
  });
});

test.each(matrix)(
  "emits correct config for target '%s' and environment '%j'",
  (target, environments) => {
    const apiMock = {
      assertVersion: version => {
        if (version !== 7) {
          throw new Error(`Only Babel API version 7 supported, given version: ${version}`);
        }
      },
    };
    expect(preset(apiMock, { target, environments })).toMatchSnapshot();
  },
);
