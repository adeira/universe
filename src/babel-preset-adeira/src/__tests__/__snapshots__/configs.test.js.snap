// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emits correct config for target 'flow' and environment '{"browsers":["last 2 versions","ie >= 11"]}' 1`] = `
{
  "presets": [],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression-check.js",
    "<BABEL_PRESET_ROOT>/src/dev-declaration.js"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "bigInt",
      "throwExpressions",
      "classPrivateProperties",
      "classProperties",
      "dynamicImport",
      "nullishCoalescingOperator",
      "numericSeparator",
      "objectRestSpread",
      "optionalChaining"
    ]
  },
  "retainLines": true
}
`;

exports[`emits correct config for target 'flow' and environment '{"node":"current"}' 1`] = `
{
  "presets": [],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression-check.js",
    "<BABEL_PRESET_ROOT>/src/dev-declaration.js"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "bigInt",
      "throwExpressions",
      "classPrivateProperties",
      "classProperties",
      "dynamicImport",
      "nullishCoalescingOperator",
      "numericSeparator",
      "objectRestSpread",
      "optionalChaining"
    ]
  },
  "retainLines": true
}
`;

exports[`emits correct config for target 'flow' and environment 'undefined' 1`] = `
{
  "presets": [],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression-check.js",
    "<BABEL_PRESET_ROOT>/src/dev-declaration.js"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "bigInt",
      "throwExpressions",
      "classPrivateProperties",
      "classProperties",
      "dynamicImport",
      "nullishCoalescingOperator",
      "numericSeparator",
      "objectRestSpread",
      "optionalChaining"
    ]
  },
  "retainLines": true
}
`;

exports[`emits correct config for target 'js' and environment '{"browsers":["last 2 versions","ie >= 11"]}' 1`] = `
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "debug": false,
        "modules": "commonjs",
        "targets": {
          "browsers": [
            "last 2 versions",
            "ie >= 11"
          ]
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression-check.js",
    "<BABEL_PRESET_ROOT>/src/dev-expression.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-invariant.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-warning.js",
    "@babel/plugin-transform-flow-strip-types",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-numeric-separator",
    [
      "@babel/plugin-transform-runtime",
      {
        "useESModules": false
      }
    ],
    "@kiwicom/babel-plugin-orbit-components"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "bigInt",
      "throwExpressions"
    ]
  },
  "retainLines": false
}
`;

exports[`emits correct config for target 'js' and environment '{"node":"current"}' 1`] = `
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "debug": false,
        "modules": "commonjs",
        "targets": {
          "node": "current"
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression-check.js",
    "<BABEL_PRESET_ROOT>/src/dev-expression.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-invariant.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-warning.js",
    "@babel/plugin-transform-flow-strip-types",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-numeric-separator",
    [
      "@babel/plugin-transform-runtime",
      {
        "useESModules": false
      }
    ],
    "@kiwicom/babel-plugin-orbit-components"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "bigInt",
      "throwExpressions"
    ]
  },
  "retainLines": false
}
`;

exports[`emits correct config for target 'js' and environment 'undefined' 1`] = `
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "debug": false,
        "modules": "commonjs",
        "targets": {
          "node": "current",
          "browsers": [
            "last 2 versions",
            "ie >= 11"
          ]
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression-check.js",
    "<BABEL_PRESET_ROOT>/src/dev-expression.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-invariant.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-warning.js",
    "@babel/plugin-transform-flow-strip-types",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-numeric-separator",
    [
      "@babel/plugin-transform-runtime",
      {
        "useESModules": false
      }
    ],
    "@kiwicom/babel-plugin-orbit-components"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "bigInt",
      "throwExpressions"
    ]
  },
  "retainLines": false
}
`;

exports[`emits correct config for target 'js-esm' and environment '{"browsers":["last 2 versions","ie >= 11"]}' 1`] = `
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "debug": false,
        "modules": false,
        "targets": {
          "browsers": [
            "last 2 versions",
            "ie >= 11"
          ]
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression-check.js",
    "<BABEL_PRESET_ROOT>/src/dev-expression.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-invariant.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-warning.js",
    "@babel/plugin-transform-flow-strip-types",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-numeric-separator",
    [
      "@babel/plugin-transform-runtime",
      {
        "useESModules": true
      }
    ],
    "@kiwicom/babel-plugin-orbit-components"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "bigInt",
      "throwExpressions"
    ]
  },
  "retainLines": false
}
`;

exports[`emits correct config for target 'js-esm' and environment '{"node":"current"}' 1`] = `
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "debug": false,
        "modules": false,
        "targets": {
          "node": "current"
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression-check.js",
    "<BABEL_PRESET_ROOT>/src/dev-expression.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-invariant.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-warning.js",
    "@babel/plugin-transform-flow-strip-types",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-numeric-separator",
    [
      "@babel/plugin-transform-runtime",
      {
        "useESModules": true
      }
    ],
    "@kiwicom/babel-plugin-orbit-components"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "bigInt",
      "throwExpressions"
    ]
  },
  "retainLines": false
}
`;

exports[`emits correct config for target 'js-esm' and environment 'undefined' 1`] = `
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "debug": false,
        "modules": false,
        "targets": {
          "node": "current",
          "browsers": [
            "last 2 versions",
            "ie >= 11"
          ]
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression-check.js",
    "<BABEL_PRESET_ROOT>/src/dev-expression.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-invariant.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-warning.js",
    "@babel/plugin-transform-flow-strip-types",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-numeric-separator",
    [
      "@babel/plugin-transform-runtime",
      {
        "useESModules": true
      }
    ],
    "@kiwicom/babel-plugin-orbit-components"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "bigInt",
      "throwExpressions"
    ]
  },
  "retainLines": false
}
`;
